go/test/endtoend/vreplication/multi_tenant_test.go

```diff
@@ -0,0 +1,298 @@
+	mtm.insertSomeData(t, tenantId, userKeyspace, 10)
+}
+
+func (mtm *multiTenantMigration) setup(tenantId int64) {
+	log.Infof("Creating MoveTables for tenant %d", tenantId)
+	mtm.setLastID(tenantId, 0)
+	sourceKeyspace := getSourceKeyspace(tenantId)
+	userKeyspace := getUserKeyspace(tenantId)
+	_, err := vc.AddKeyspace(mtm.t, []*Cell{vc.Cells["zone1"]}, sourceKeyspace, "0", mtVSchema, mtSchema, 1, 0, int(1000+tenantId*100), nil)
+	require.NoError(mtm.t, err)
+	updateKeyspaceRoutingRules(mtm.t, vc, userKeyspace, sourceKeyspace)
+	mtm.initTenantData(mtm.t, tenantId, userKeyspace)
+}
+
+func (mtm *multiTenantMigration) start(tenantId int64) {
+	sourceKeyspace := getSourceKeyspace(tenantId)
+	userKeyspace := getUserKeyspace(tenantId)
+	mtm.setTenantMigrationStatus(tenantId, tenantMigrationStatusMigrating)
+	mt := newVtctldMoveTables(&moveTablesWorkflow{
+		workflowInfo: &workflowInfo{
+			vc:             vc,
+			workflowName:   fmt.Sprintf("wf_tenant%d", tenantId),
+			targetKeyspace: mtm.targetKeyspace,
+		},
+		sourceKeyspace: sourceKeyspace,
+		tables:         mtm.tables,
+		createFlags: []string{
+			"--additional-filter", fmt.Sprintf("%s=%d", mtm.tenantIdColumnName, tenantId),
+			"--use-keyspace-routing-rules",
+			"--source-keyspace-alias", userKeyspace,
+		},
+	})
+	mtm.setActiveMoveTables(tenantId, mt)
+	mt.Create()
+}
+
+func (mtm *multiTenantMigration) insertSomeData(t *testing.T, tenantId int64, userKeyspace string, numRows int64) {
+	vtgateConn, closeConn := getVTGateConn()
+	defer closeConn()
+	idx := mtm.getLastID(tenantId)
+	for i := idx + 1; i <= idx+numRows; i++ {
+		execVtgateQuery(t, vtgateConn, "",
+			fmt.Sprintf("insert into %s.t1(id, tenant_id) values(%d, %d)", userKeyspace, i, tenantId))
+	}
+	mtm.setLastID(tenantId, idx+numRows)
+}
+
+func (mtm *multiTenantMigration) switchTraffic(tenantId int64) {
+	t := mtm.t
+	userKeyspace := getUserKeyspace(tenantId)
+	mt := mtm.activeMoveTables[tenantId]
+	ksWorkflow := fmt.Sprintf("%s.%s", mtm.targetKeyspace, mt.workflowName)
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Running.String())
+	mtm.insertSomeData(t, tenantId, userKeyspace, 10)
+	mt.SwitchReadsAndWrites()
+	//printKeyspaceRoutingRules(t, vc, "After MoveTables SwitchTraffic")
+	mtm.insertSomeData(t, tenantId, userKeyspace, 10)
+}
+
+func (mtm *multiTenantMigration) complete(tenantId int64) {
+	mt := mtm.activeMoveTables[tenantId]
+	mt.Complete()
+	vtgateConn := vc.GetVTGateConn(mtm.t)
+	defer vtgateConn.Close()
+	waitForQueryResult(mtm.t, vtgateConn, "",
+		fmt.Sprintf("select count(*) from multi_tenant.t1 where tenant_id=%d", tenantId),
+		fmt.Sprintf("[[INT64(%d)]]", mtm.getLastID(tenantId)))
+}
+
+const waitTimeout = 10 * time.Minute
+
+func TestMultiTenant(t *testing.T) {
+	setSidecarDBName("_vt")
+	origDefaultRdonly := defaultRdonly
+	defer func() {
+		defaultRdonly = origDefaultRdonly
+	}()
+	defaultRdonly = 0
```

Comment:
A comment here will be useful. I found this package level var definition and understood that it means we don't run any RDONLY tablets.